### PR TITLE
*: enable errcheck for nogo

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: ref #35345

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
fail to build if remove any ```//nolint: errcheck``` 
- [ ] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by tangenta** [comment]
Can we analyze and ignore the "xxx.Close()" related errcheck?

**Review by tangenta — APPROVED** [review]

**Comment by hawkingrei** [comment]
> Can we analyze and ignore the "xxx.Close()" related errcheck?

I add the whitelist to it so that ```xxx.Close()``` and ```xxx.Fprintxx``` will be ignored.

**Review by tisonkun — APPROVED** [review]
Generally looks good. Minor suggestion inline.

File: br/pkg/storage/writer.go [hunk 1]
```diff
@@ -159,6 +159,7 @@ func (u *bufferedWriter) Write(ctx context.Context, p []byte) (int, error) {
 				continue
 			}
 		}
+		//nolint: errcheck
 		u.buf.Flush()
```
[the remark this answers is not in the document]
Comment (hawkingrei) [reply br/pkg/storage/writer.go:163]: You are right.

**Pull request merged**